src/network/networkctl.c

```diff
@@ -470,42 +478,46 @@ static int dump_gateways(
                 if (r < 0)
                         log_debug_errno(r, "Could not get description of gateway: %m");
 
-                printf("%*s%s",
-                       (int) strlen(prefix),
-                       i == 0 ? prefix : "",
-                       gateway);
-
-                if (description)
-                        printf(" (%s)", description);
+                if (description) {
+                        with_description = strjoin(gateway, " (", description, ")");
+                        if (!with_description)
+                                return -ENOMEM;
+                }
 
                 /* Show interface name for the entry if we show
                  * entries for all interfaces */
                 if (ifindex <= 0) {
                         char name[IF_NAMESIZE+1];
 
                         if (if_indextoname(local[i].ifindex, name)) {
-                                fputs(" on ", stdout);
-                                fputs(name, stdout);
-                        } else
-                                printf(" on %%%i", local[i].ifindex);
+                                r = table_add_string_cell(table, NULL, "%s on %s", with_description ?: gateway, name);
+                                if (r < 0)
+                                        return r;
+                        } else {
+                                r = table_add_string_cell(table, NULL, "%s on %%%i", with_description ?: gateway, local[i].ifindex);
+                                if (r < 0)
+                                        return r;
```

Comment:
the `if (r < 0) return r;` part you could move outside of the if block and thus share between both branches